cli/GNUmakefile

```diff
@@ -0,0 +1,140 @@
+MAKE_MAJOR_VER    := $(shell echo $(MAKE_VERSION) | cut -d'.' -f1)
+
+ifneq ($(shell test $(MAKE_MAJOR_VER) -gt 3; echo $$?),0)
+$(error Make version $(MAKE_VERSION) is not supported, please install GNU Make 4.x)
+endif
+
+GREP                      ?= $(shell command -v ggrep 2> /dev/null || command -v grep 2> /dev/null)
+SED                       ?= $(shell command -v gsed 2> /dev/null || command -v sed 2> /dev/null)
+AWK                       ?= $(shell command -v gawk 2> /dev/null || command -v awk 2> /dev/null)
+RUSTUP                    ?= $(shell command -v rustup 2> /dev/null)
+RPM_ARCH                  := $(shell uname -m)
+VERSION                   ?= $(shell $(GREP) -Po '^version\s+=\s+"\K.*?(?=")' $(CURDIR)/unitctl/Cargo.toml)
+SRC_REPO                  := https://github.com/nginxinc/unit-rust-sdk
+DEFAULT_TARGET            ?= $(shell $(RUSTUP) toolchain list | $(GREP) '(default)' | cut -d' ' -f1 | cut -d- -f2-)
+SHELL                     := /bin/bash
+OUTPUT_BINARY             ?= unitctl
+PACKAGE_NAME              ?= unitctl
+CARGO                     ?= cargo
+DOCKER                    ?= docker
+DOCKER_BUILD_FLAGS        ?= --load
+CHECKSUM                  ?= sha256sum
+OPENAPI_GENERATOR_VERSION ?= 6.6.0
+COMMITSAR_DOCKER          := $(DOCKER) run --tty --rm --workdir /src -v "$(CURDIR):/src" aevea/commitsar
+COMMITSAR                 ?= $(shell command -v commitsar 2> /dev/null)
+
+# Define platform targets based off of the current host OS
+# If we are running MacOS, then we can build for MacOS platform targets that have been installed in rustup
+# If we are running Linux, then we can build for Linux platform targets that have been installed in rustup
+UNAME = $(shell uname | tr '[:upper:]' '[:lower:]')
```

Comment:
`uname -s`

Reply:
Yeah, you caught one of my historical cross-unix compatibility hacks. `uname -s` is fine.